### ISSUE TITLE
Login in with the desktop client when using SSO (SAML with Keycloak) fails

### ISSUE THREAD

**Issue by lucode** [nextcloud/desktop#1084]
We run a Nectcloud instance on Hetzner and using Keycloak ID server witch allows SSO with SAML.
On the browser everything works great, but we can’t login into Nextcloud with the Desktop Client.
Android Client works too, but with the Desktop client the process stucks when I want to give Access to the files. We are not the only ones witch struggle with this issue:
https://help.nextcloud.com/t/issue-login-in-with-the-desktop-client-when-using-sso-saml-with-keycloak/47063

### Expected behaviour
The Client should at this stage connect to the server an start synchronzing the files.

### Actual behaviour
(https://help.nextcloud.com/uploads/default/original/2X/7/7675f215c2f638e02511e93f0d3ae79599a4d726.png)
The client shows then this state forever. The same siutation is when I using the app token instead of password. When I look in my settings I see there both Desktop and Android client are connected properly, so I assume the error is neither at the server or the Keycloak side. It must be a Desktop Client issue.

### Steps to reproduce
Same issue on different machines (Win & Ubuntu)

### Client configuration
Client version:

I have installed client Windows version: Version 2.5.1final (build 20181204).
And Ubuntu ppa version 2.5.1. It is the same issue.

Operating system:
Win 10
OS language:
German and EN

Installation path of client:
C:\Program Files (x86)\Nextcloud\nextcloud.exe

### Server configuration
Hetzner Owncloud SaaS

Nextcloud version:
15.0.2

### Logs
I don't get any logs, because the client is not yet running.

2. Web server error log:
not accessible
3. Server logfile: nextcloud log (data/nextcloud.log):
No server logs

**Comment by duckdiver** [comment]
in our scenario, when we try to Login via NC DektopClient V2.51 and SSO & SAML, I get the SSO Authentication Login Screen for User and PW Input:
https://paste.pics/cc6ca7340737e07c885c21ba32adf2ce

https://paste.pics/38c6db52f1c2e47424508f7fca7da567

After that we stuck with that screen
https://paste.pics/cc6ca7340737e07c885c21ba32adf2ce

**Comment by duckdiver** [comment]
in our scenario, when we try to Login via NC DektopClient V2.51 and SSO & SAML, I get the SSO Authentication Login Screen for User and PW Input:
https://paste.pics/13434e7ad815a98fb4fd93ae320819a6
https://paste.pics/38c6db52f1c2e47424508f7fca7da567

After that we stuck with that screen
https://paste.pics/cc6ca7340737e07c885c21ba32adf2ce

**Comment by lucode** [comment]
@duckdiver that is another issue. your issue happens before the client wants to start to sync. Login works basically at our scenario.

**Comment by duckdiver** [comment]
with NC Server V13

**Comment by duckdiver** [comment]
@lucode dont you stuck already in the first screen?

**Comment by lucode** [comment]
No at the last one: 

> When I look in my settings I see there both Desktop and Android client are connected properly,

**Comment by lucode** [comment]
From my point of view you did mix up some configs about using token or password.

**Comment by duckdiver** [comment]
No, 
but the credentials are not send within Mozilla-APP with NC Desktop CLient.
When using Login in IE Webbrowser it works perfectly.
Also with Mobile App.

**Comment by mudasaryasin** [comment]
Guys, any work around, i am facing this issue even i login successfully from IE.

**Comment by mudasaryasin** [comment]
Here are logs on client 

```
OCC::PropfindJob::finished 	*not* successful, http result code is 302 "https://mync.xxxxx.co/index.php/apps/user_saml/saml/selectUserBackEnd?redirectUrl="
[OCC::OwmyncSetupWizard::slotAuthError 	Authed request was redirected to "https://mync.xxxxx.co/index.php/apps/user_saml/saml/selectUserBackEnd?redirectUrl="
[OCC::WebViewPageUrlSchemeHandler::requestStarted 	Got user:  "YMudasar" , server:  "https://mync.xxxxx.co"
[OCC::WebViewPage::urlCatched 	Got user:  "YMudasar" , server:  "https://mync.xxxxx.co"
[OCC::WebViewPage::urlCatched 	URL:  "https://mync.xxxxx.co"
[OCC::OwmyncSetupWizard::slotConnectToOCUrl 	Connect to url:  "https://mync.xxxxx.co"
[OCC::WebFlowCredentials::createQNAM 	Get QNAM
[OCC::AccessManager::createRequest 	6 "PROPFIND" "https://mync.xxxxx.co/remote.php/webdav/" has X-Request-ID "xxxxx-6ac2-4efd-9147-xxxxxxx"
[OCC::AbstractNetworkJob::start 	OCC::PropfindJob created for "https://mync.xxxxx.co" + "/" "OCC::OwmyncSetupWizard"
[OCC::WebFlowCredentials::slotFinished 	request finished
[OCC::WebFlowCredentials::stillValid 	Still valid?
[OCC::WebFlowCredentials::stillValid 	QNetworkReply::NetworkError(NoError)
[OCC::WebFlowCredentials::stillValid 	"Unknown error"
```

**Comment by mudasaryasin** [comment]
Guys any clue?

**Comment by lucode** [comment]
I got the log now:
```
[OCC::AbstractNetworkJob::start 	OCC::PropfindJob created for "https://nc.cooby.org" + "/" "OCC::OwncloudSetupWizard"
[OCC::WebFlowCredentials::slotFinished 	request finished
[OCC::WebFlowCredentials::stillValid 	Still valid?
[OCC::WebFlowCredentials::stillValid 	QNetworkReply::NetworkError(NoError)
[OCC::WebFlowCredentials::stillValid 	"Unbekannter Fehler"
[OCC::PropfindJob::finished 	PROPFIND of QUrl("https://nc.cooby.org/remote.php/webdav/") FINISHED WITH STATUS "OK"
[OCC::PropfindJob::finished 	*not* successful, http result code is 302 "http://nc.cooby.org/apps/user_saml/saml/login?originalUrl=&idp=1&requesttoken=7cmxYSv2hqN03nDPJGkOqtDvhnp/iEm/ZrCQzDBAbkk%3D%3AjPDwUxqv4ZocpDqFEztW8JLXxS4axgXLKf/cgF8rFic%3D"
[OCC::OwncloudSetupWizard::slotAuthError 	Authed request was redirected to "http://nc.cooby.org/apps/user_saml/saml/login?originalUrl=&idp=1&requesttoken=7cmxYSv2hqN03nDPJGkOqtDvhnp/iEm/ZrCQzDBAbkk%3D%3AjPDwUxqv4ZocpDqFEztW8JLXxS4axgXLKf/cgF8rFic%3D"
```
Is on another server with NC 15.0.2 installed.

**Comment by mudasaryasin** [comment]
For me it is working fine after disabling this option in SSO and SAML plugin 

> "Use SAML auth for the ibex Nextcloud desktop clients (requires user re-authentication)"

**Comment by lucode** [comment]
@mudasaryasin you are our hero, your suggestion works for us too.
The question remains if this is a bug or a feature?
![grafik](https://user-images.githubusercontent.com/2928740/54738887-5b55ee00-4bb6-11e9-8da3-0682b8119a58.png)

**Comment by mudasaryasin** [comment]
No me, it was discover by my colleague.

**Comment by rullzer** [comment]
It is kind of a bug. But also a feature. The issue is that the old clients handled saml internally. Which often did :boom: and caused relogins to happen all the time.

There is a bug somewhere in the saml detection logic. But I can't figure out where since I'm unable to reproduce it myself.

**Comment by olivierb2** [comment]
+1

**Comment by holger13** [comment]
+3

**Comment by Yzed** [comment]
seeing same with Mac Desktop Client:

first the 302 redirect, then next request gets a "Method not allowed" 405:

`PROPFIND /apps/user_saml/saml/login?originalUrl=&[...]` 

NGINX is configured to allow PROPFIND

waiting a few sec's actually Desktop Client seem to continue and is successfully logged in and starts syncing... it appears the initial login gets a 405 only.

**Comment by pelzvieh** [comment]
Same here, same fix worked.

**Comment by wget** [comment]
Hello everyone,
I confirm this is fixing the issue we had at The Document Foundation. I think we can close now.
https://github.com/nextcloud/desktop/issues/830#issuecomment-449589408

**Comment by camilasan** [comment]
We fixed it with the new login flow in 2.6.
